### PR TITLE
Implement optionSelected attribute

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #97 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

* Implement `optionSelected` attribute to `auro-radio-group`
* Refactor doc examples

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: apiExamples/basicGroup.html [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <auro-radio-group>
-  <span slot="legend">Form label goes here</span>
-  <auro-radio id="radio28" label="Yes" name="radioDemo" value="yes"></auro-radio>
-  <auro-radio id="radio29" label="No" name="radioDemo" value="no"></auro-radio>
-  <auro-radio id="radio30" label="Maybe" name="radioDemo" value="maybe"></auro-radio>
+  <span slot="legend">Are you willing to volunteer resources?</span>
+  <auro-radio id="radio4" label="Yes" name="radioDemo" value="yes">Volunteering is AWESOME!</auro-radio>
+  <auro-radio id="radio5" label="No" name="radioDemo" value="no">Sorry, I do not have the time or resources</auro-radio>
+  <auro-radio id="radio6" label="Maybe" name="radioDemo" value="maybe">Please send me more information</auro-radio>
```
Comment (jason-capsule42) [inline apiExamples/basicGroup.html:5]: Why did all the examples content change? It doesn't seem connected to the change for `optionSelected`.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

**Comment by jordanjones243** [comment]
PR will be addressed once the form validation PR has been merged into the RC branch:
- https://github.com/AlaskaAirlines/auro-radio/pull/129

File: test/auro-radio.test.js [hunk 2]
```diff
@@ -112,4 +112,20 @@ describe('auro-radio', () => {
...
+        <auro-radio id="radio2" label="No" name="radioDemo" value="no"></auro-radio>
+        <auro-radio id="radio3" label="Maybe" name="radioDemo" value="maybe"></auro-radio>
+      </auro-radio-group>
+    `);
+
+    const radioButtons = [...el.querySelectorAll('auro-radio')];
+    const selected = radioButtons.find(radioButton => radioButton.hasAttribute('checked'));
+
+    await expect(el.optionSelected).to.equal(selected);
+  });
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply test/auro-radio.test.js:130]: Currently, editing the `value` attribute does not affect the state of anything in the radio group, so I am not exactly sure what you are aiming to test. If you feel that you are on the track of something important, we can talk about it further. Keep in mind that this work will be released alongside the form validation work.

`value` attribute setting can be addressed in this ticket:
- https://github.com/AlaskaAirlines/auro-radio/issues/130

File: docs/partials/api.md [hunk 3]
```diff
@@ -79,21 +114,19 @@ The `error` attribute used to set an error state on the entire `<auro-radio-grou
 
 </auro-accordion>
 
-### horizontal
-
-Using the `horizontal` attribute will render the `auro-radio` elements in a horizontal line.
+#### Single
 
-**Note**: Using the horizontal attribute has a limit of 3 options. Beyond three, options will be listed in vertically.
+The `error` attribute used to set an error state on a single `<auro-radio>` element.
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply docs/partials/api.md:119]: The example is removed in the form validation repo.

File: test/auro-radio.test.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jason-capsule42) [reply test/auro-radio.test.js]: Resolving based on the future work ticket

**Review by jason-capsule42 — APPROVED** [review]

**Pull request merged**